Mathlib/MeasureTheory/Integral/IntegrableOn.lean

```diff
@@ -89,19 +89,24 @@ def IntegrableOn (f : α → ε) (s : Set α) (μ : Measure α := by volume_tac)
 theorem IntegrableOn.integrable (h : IntegrableOn f s μ) : Integrable f (μ.restrict s) :=
   h
 
+variable [TopologicalSpace ε'] [ENormedAddMonoid ε']
+
 @[simp]
-theorem integrableOn_empty : IntegrableOn f ∅ μ := by simp [IntegrableOn, integrable_zero_measure]
+theorem integrableOn_empty : IntegrableOn f ∅ μ := by
+  simp [IntegrableOn]
 
 @[simp]
 theorem integrableOn_univ : IntegrableOn f univ μ ↔ Integrable f μ := by
   rw [IntegrableOn, Measure.restrict_univ]
 
-theorem integrableOn_zero : IntegrableOn (fun _ => (0 : E)) s μ :=
+theorem integrableOn_zero : IntegrableOn (fun _ => (0 : ε')) s μ :=
   integrable_zero _ _ _
 
 @[simp]
-theorem integrableOn_const {C : E} : IntegrableOn (fun _ => C) s μ ↔ C = 0 ∨ μ s < ∞ :=
-  integrable_const_iff.trans <| by rw [isFiniteMeasure_restrict, lt_top_iff_ne_top]
+theorem integrableOn_const {C : ε'} (hC : ‖C‖ₑ ≠ ∞ := by finiteness) :
```

Comment:
The current version forces you to use `(integrableOn_const).2` to make sure the autoparam fires. What about renaming this one to `integrableOn_const_iff`, and having the most typical use case as `integrableOn_const`, phrased as
```
theorem integrableOn_const {C : ε'} (hs : μ s ≠ ∞ := by finiteness) (hC : ‖C‖ₑ ≠ ∞ := by finiteness) : IntegrableOn (fun _ ↦ C) s μ
```

Reply:
That's a very reasonable idea, let me implement that.

Reply:
Done, and mathlib builds.